### PR TITLE
[android] Reset room when JS thread reloads while on call

### PR DESCRIPTION
**Pull request by marqroldan**
Explained here https://github.com/blackuy/react-native-twilio-video-webrtc/issues/559

tl;dr title + `componentWillUnmount` doesn't get called when you restart the JS thread 

This PR updates the existing `releaseResource()` to be also used inside the constructor and `onHostDestroy`

### Behavior in `master` branch

1. When participants are in the room for the first time

2. When one participant have their JS thread reload and attempt to join again

### With the PR
1. Same with no. 1 above,
2. LocalView and participant feed can now be seen again

Notes:
- Formatted with Android studio

Question/Concern:
- I am not sure if the mediaListeners for the remote participants are automatically cleared? I believe they are when you disconnect from the room
- There might be cases where we want the room to still be connected, but I'm not sure how to properly handle this
  -  For example launching a native UI when you receive a call, and then continuing in the JS thread.
     -  Maybe such functionality can be added in the future by exposing a boolean or something (not an Android developer hehe)

### CODE REVIEW

**Comment by marqroldan** [comment]
Rebased the PR with master, changes are clearer now :D

File: android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java [hunk 1]
```diff
@@ -208,8 +208,56 @@
...
+         */
+        if (room != null && room.getState() != Room.State.DISCONNECTED) {
+            room.disconnect();
+            disconnectedFromOnDestroy = true;
+        }
+        room = null;
+
+
+        if (localParticipant != null) {
+            localParticipant.unpublishTrack(localVideoTrack);
```
Comment (slycoder) [inline android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java:235]: Do you need that localVideoTrack is not null?

**Review by slycoder — APPROVED** [review]
Just one comment, otherwise lgtm!
